internal/bodymasking/bodymasking.go

```diff
@@ -0,0 +1,55 @@
+package bodymasking
+
+import (
+	"errors"
+	"fmt"
+	"mime"
+	"regexp"
+
+	"github.com/speakeasy-api/speakeasy-go-sdk/internal/utils"
+)
+
+var (
+	ErrMaskNumberTypeError = errors.New("can't mask a number with a string")
+	ErrMaskStringTypeError = errors.New("can't mask a string with a number")
+)
+
+const (
+	stringFieldMatchRegex = `("%s": *)(".*?[^\\]")( *[, \n\r}]?)`
+	numberFieldMatchRegex = `("%s": *)(-?[0-9]+\.?[0-9]*)( *[, \n\r}]?)`
+)
+
+func MaskBodyRegex(body, mimeType string, stringMasks map[string]string, numberMasks map[string]string) (string, error) {
+	mediaType, _, err := mime.ParseMediaType(mimeType)
+	if err != nil {
+		return "", err
+	}
+
+	if mediaType != "application/json" {
```

Comment:
Should there be a callout in the readme or something that this is only set to work with json for the time being?

Reply:
done